edi-module/error.bal

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents EDI module related errors
+public type Error distinct error;
```

Comment:
Since we have only one error and error is also type, we can move this to edi_types.bal too.

Reply:
edi_types.bal contains types required to represent EDI segments, fields, etc. So added the error type to the edi_translator.bal.